### PR TITLE
feat: swap from pool

### PR DESCRIPTION
**Pull request by agualis**
This can me merged hiding the pool advanced options first.

swapPool

TODO: 
- The swap pool feature is not yet available in the SDK
- Test edge-cases like nested pools

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,7 +0,0 @@
-{
-  "eslint.workingDirectories": [
```
Comment (agualis) [inline .vscode/settings.json:2]: @garethfuller deleted this file from git cache as now it is .gitignored
But it would be good to use .editorconfig or similar.

File: packages/lib/modules/pool/PoolDetail/PoolHeader/PoolHeader.tsx [hunk 2]
```diff
@@ -55,14 +56,18 @@ export function PoolHeader() {
...
+              onClick={handleClick}
+              variant="primary"
+              size="lg"
+              isDisabled={isAddLiquidityBlocked}
+              w="full"
+            >
+              Add liquidity
+            </Button>
+            <PoolAdvancedOptions />
+          </HStack>
```
Comment (garethfuller) [inline packages/lib/modules/pool/PoolDetail/PoolHeader/PoolHeader.tsx:70]: Just to be clear, you are going to hide this before merging?

File: packages/lib/modules/swap/useIsPoolSwap.tsx [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { usePathname } from 'next/navigation'
+
+export function useIsPoolSwap() {
+  const pathname = usePathname()
+  return pathname.includes('/pools') && pathname.includes('/swap')
+}
```
Comment (garethfuller) [inline packages/lib/modules/swap/useIsPoolSwap.tsx:6]: What happens if we allow pool swaps directly in the swap UI eventually?

File: packages/lib/modules/swap/SwapForm.tsx [hunk 4]
```diff
@@ -94,6 +102,22 @@ export function SwapForm() {
...
+      tokens.length === 2 &&
+      tokenSelectKey === 'tokenIn' &&
+      isSameAddress(token.address, tokenOut.address)
+    )
+      return switchTokens()
+    if (
+      tokens.length === 2 &&
+      tokenSelectKey === 'tokenOut' &&
+      isSameAddress(token.address, tokenIn.address)
+    )
```
Comment (garethfuller) [inline packages/lib/modules/swap/SwapForm.tsx:116]: Shouldn't this be handled in the same we were handling it in the swap UI? i.e. you just can't select the same token that's currently selected?

Comment (garethfuller) [reply packages/lib/modules/swap/SwapForm.tsx:116]: Seems like it should be logic that's handled in the modal, not here.

File: packages/lib/modules/pool/PoolDetail/PoolHeader/PoolHeader.tsx (hunk 2 above)
Comment (agualis) [reply packages/lib/modules/pool/PoolDetail/PoolHeader/PoolHeader.tsx:70]: Yes

File: packages/lib/modules/swap/SwapProvider.tsx [hunk 5]
```diff
@@ -60,6 +61,8 @@ export type PathParams = {
   amountOut?: string
   // When urlTxHash is present the rest of the params above are not used
   urlTxHash?: Hash
+  // Only used by pool swap
+  poolTokens?: GqlToken[]
```
[the remark this answers is not in the document]
Comment (agualis) [reply packages/lib/modules/swap/SwapProvider.tsx:65]: I refactored the types so it's clearerer that `poolTokens` do not belong to the path

File: packages/lib/modules/swap/useIsPoolSwap.tsx (hunk 3 above)
Comment (agualis) [reply packages/lib/modules/swap/useIsPoolSwap.tsx:6]: Refactored into `isPoolSwap` + `useIsPoolSwapUrl`

File: packages/lib/modules/swap/SwapForm.tsx (hunk 4 above)
Comment (agualis) [reply packages/lib/modules/swap/SwapForm.tsx:118]: I wanted to reuse the same selector that we are using to choose between native asset and wrapped native asset where you can chose the currently selected token.

The logic is handling the case when you chose the other token in a 2-token pool swap. So instead of disabling the other token, it switches them:

https://github.com/user-attachments/assets/259e5c66-1c9a-4abc-a393-975947dc2ea1

File: packages/lib/modules/swap/useIsPoolSwapUrl.tsx [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import { usePathname } from 'next/navigation'
+
+export function useIsPoolSwapUrl() {
+  const pathname = usePathname()
+  return pathname.includes('/pools') && pathname.includes('/swap')
+}
```
Comment (garethfuller) [inline packages/lib/modules/swap/useIsPoolSwapUrl.tsx:6]: I don't think this is a good approach. I was just trying to point out that the logic using the URL might not be the best way to decide if it's a pool swap. Now, if we decide to support pool swaps in the swap UI you'll have to have two conditionals all over the place.

The alternative approach would be to maintain a `isPoolSwap` state variable in the SwapProvider. The default can be set by a useSwap param (i.e. a param passed into the provider). So the swap layout used in the pool context would take a prop like `isPoolSwap` which is then passed into the provider. Then in the future we can easily switch on this flag with UI changes in the swap form.

**Review by garethfuller — CHANGES_REQUESTED** [review]

File: packages/lib/modules/swap/useIsPoolSwapUrl.tsx (hunk 6 above)
Comment (agualis) [reply packages/lib/modules/swap/useIsPoolSwapUrl.tsx:6]: We already have the `isPoolSwap` into the provider. 
But we do need the two conditionals. 

`isPoolSwap` is used to display the specific pool swap parts

`useIsPoolSwapUrl` is used to detect the specific case where the swap is happening within a pool actions layout, meaning that receipt page and redirects work in a different way

Comment (garethfuller) [reply packages/lib/modules/swap/useIsPoolSwapUrl.tsx:6]: Ok, maybe let's see.

**Review by garethfuller — CHANGES_REQUESTED** [review]
Hide advanced pool options for production.
